src/mtype.d

```diff
@@ -8089,7 +8089,12 @@ extern (C++) final class TypeStruct : Type
     {
         //printf("TypeStruct::semantic('%s')\n", sym.toChars());
         if (deco)
+        {
+            if (sc.cppmangle != CPPMANGLE.def && this.cppmangle == CPPMANGLE.def)
+                this.cppmangle = sc.cppmangle;
+            assert(sc.cppmangle == CPPMANGLE.def || this.cppmangle == sc.cppmangle);
```

Comment:
I adopted it - the condition is less complex which makes it easier to understand.